src/compiler/checker.ts

```diff
@@ -25740,7 +25752,15 @@ namespace ts {
                         relatedInfo = suggestion.valueDeclaration && createDiagnosticForNode(suggestion.valueDeclaration, Diagnostics._0_is_declared_here, suggestedName);
                     }
                     else {
-                        errorInfo = chainDiagnosticMessages(elaborateNeverIntersection(errorInfo, containingType), Diagnostics.Property_0_does_not_exist_on_type_1, declarationNameToString(propNode), typeToString(containingType));
+                        const property = declarationNameToString(propNode);
+                        const container = typeToString(containingType);

```

Comment:
As you can see below, this forces you to reparse. You already have the semantic structure, reuse that.